tests/services/phase.service.spec.ts

```diff
@@ -139,14 +142,62 @@ describe('PhaseService', () => {
   });
 
   describe('.reset()', () => {
-    beforeEach(() => {
-      phaseService = new PhaseService(githubServiceSpy, repoUrlCacheServiceSpy, loggingServiceSpy);
-    });
-
     it('should reset the currentPhase of the PhaseService', () => {
       phaseService.currentPhase = Phase.activityDashboard;
       phaseService.reset();
       expect(phaseService.currentPhase).toBe(Phase.issuesViewer);
     });
   });
+
+  describe('setupFromUrl(url)', () => {
+    it('should set items in local storage if url is valid', async () => {
+      const validUrl = `/issuesViewer?repo=${WATCHER_REPO.owner}%2F${WATCHER_REPO.name}`;
+      const localStorageSetItemSpy = spyOn(window.localStorage, 'setItem');
+
+      await phaseService.setupFromUrl(validUrl).toPromise();
+
+      expect(localStorageSetItemSpy).toHaveBeenCalledWith(STORAGE_KEYS.ORG, WATCHER_REPO.owner);
+      expect(localStorageSetItemSpy).toHaveBeenCalledWith(STORAGE_KEYS.DATA_REPO, WATCHER_REPO.name);
+    });
+
+    it('should throw error for url without repo paramater', (done) => {
+      const urlWithoutRepo = '/issuesViewer';
+
+      phaseService.setupFromUrl(urlWithoutRepo).subscribe({
+        error: (err) => {
+          expect(err).toEqual(new Error(ErrorMessageService.invalidUrlMessage()));
+          done();
+        }
+      });
+    });
+
+    it('should throw error for empty url', (done) => {
+      const emptyUrl = '';
+
+      phaseService.setupFromUrl(emptyUrl).subscribe({
+        error: (err) => {
+          expect(err).toEqual(new Error(ErrorMessageService.invalidUrlMessage()));
+          done();
+        }
+      });
+    });
+
+    it('should throw error for url with invalid repo format', (done) => {
+      const urlWithInvalidRepoFormat = '/issuesViewer?repo=InvalidRepo';
+
+      phaseService.setupFromUrl(urlWithInvalidRepoFormat).subscribe({
+        next: (value) => {
+          fail('Expected an error to be thrown');
+          done();
+        },
+        complete: () => {
+          fail('Expected an error to be thrown');
+          done();
+        },
+        error: (err) => {
```

Comment:
Wondering if we should test if this throws the `ErrorMessageService.repositoryNotPresentMessage` error instead of a generic error.

Reply:
Yes, I think it's a good idea to test for the specific error.